docs/shrc.md

```diff
@@ -66,43 +140,138 @@ The [-Xshareclasses](xshareclasses.md) option is highly configurable, allowing y
 
     In addition, if you have VMs from different Java installations, of the same Java release and installed by the same user, each VM checks whether the existing default shared cache in the cache directory is from the same Java installation as the VM. If not, the VM deletes that shared cache, then creates a new one. Specifying a different cache directory for each Java installation avoids this situation.
 
-
-
 - Ensure that the cache directory permissions are set appropriately ([`-Xshareclasses:cacheDirPerm`](xshareclasses.md#cachedirperm)).
 
-    It is good practice to explicitly set permissions for the cache directory when the defaults are not appropriate.
+    It is good practice to explicitly set permissions for the cache directory when the defaults are not appropriate. Access is controlled by operating system permissions and Java security permissions; read/write access is the default and gives all users equal rights to update the cache. On Unix systems, you can use the [`-Xshareclasses:groupAccess`](xshareclasses.md#groupaccess) suboption to allow read/write permissions for groups as well as users.
+
+    On z/OS, a cache can be accessed only by a VM that is running in the same storage key as the VM that created the cache. If the keys do not match, permission to access the cache is denied.
 
 - Set the [`-Xshareclasses:nonfatal`](xshareclasses.md#nonfatal) option.
 
-    This option means that your application can start even if there is a problem opening or creating the shared cache, in which case, the VM might be able to start *without* class data sharing.
+    Setting this option allows your application to start even if there is a problem opening or creating the shared cache. In this situation, the VM might be able to start *without* class data sharing.
+
+When the cache is active, you can adjust the soft maximum size by using the [`-Xshareclasses:adjustsoftmx`](xshareclasses.md#adjustsoftmx) cache utility.
+
+Using the [`-Xshareclasses:listAllCaches`](xshareclasses.md#listallcaches) cache utility allows you to identify all the caches that exist on a system, including compatible and incompatible caches.
```

Comment:
This should be another bullet point. Also it doesn't find all caches on a system, only the ones in the default directory, or the ones in the directory specified by adding a `cacheDir` suboption.

Reply:
I think I worked this in ok by moving it to the top, but let me know if you have an alternative suggestion: 

*-  Before starting your application, use the [`-Xshareclasses:listAllCaches`](xshareclasses.md#listallcaches) cache utility to identify a suitable cache that your application might share.*
- *This option lists all the caches that exist in the default directory, including compatible and incompatible caches. You can also specify the `cacheDir` suboption to look for caches in a specified directory.*

Then changed the lead in to the next bullet:

*- If you are creating a new cache, .....*

Reply:
I don't think we should say " to identify a suitable cache that your application might share". It's best if every application has it's own cache. There should be a point for that somewhere. `listAllCaches` is for maintenance, such as to find a delete obsolete caches.

Reply:
Updated as follows:

*- Before starting your application, use the [`-Xshareclasses:listAllCaches`](xshareclasses.md#listallcaches) cache utility **to review and maintain the existing caches on your system**.*

 - *This option lists all the caches that exist in the default directory, including compatible and incompatible caches. You can also specify the `cacheDir` suboption to look for caches in a specified directory. **Remove any obsolete caches, as described in [Housekeeping](#housekeeping)**.*